src/hidimstat/ensemble_clustered_inference.py

```diff
@@ -1,45 +1,330 @@
 import numpy as np
+from sklearn.base import clone
+from sklearn.utils import resample
 from joblib import Parallel, delayed
 
-from .clustered_inference import clustered_inference
-from .multi_sample_split import aggregate_medians, aggregate_quantiles
+from hidimstat.desparsified_lasso import (
+    desparsified_lasso,
+    desparsified_lasso_pvalue,
+    desparsified_group_lasso_pvalue,
+)
+from hidimstat.stat_tools import aggregate_quantiles
 
 
-def _ensembling(
-    list_beta_hat,
-    list_pval,
-    list_pval_corr,
-    list_one_minus_pval,
-    list_one_minus_pval_corr,
-    method="quantiles",
-    gamma_min=0.2,
+def _subsampling(n_samples, train_size, groups=None, seed=0):
```

Comment:
this function should probably go to another module.

Reply:
This function is used in this model; I don't see the interest in moving into another file.

Reply:
Ok, I will move it to _utils.

Reply:
done